tests/evaluation/test_common.py

```diff
@@ -0,0 +1,33 @@
+import numpy as np
+import pytest
+
+from orion.evaluation.common import _any_overlap, _overlap, _overlap_segment
+
+
+@pytest.fixture
+def expected():
+    return (1, 5)
+
+
+@pytest.fixture
+def observed():
+    return (2, 5)
+
+
+def test__overlap(expected, observed):
```

Comment:
Are these tests enough?

Even though the test function is quite simple, I would I feel that here we would want a few test cases, especially the edge ones:
- no overlap
- no overlap, edges touch
- exact match
- one contained within the other, one edge matches
- one fully contained within the other, no edge match
- start == end (if this is something that could happen - current code seems to fail at this one)

Reply:
The same thing applies for the other functions being tested in this module.

Reply:
Updated the unittests to include edge cases and edge cases with points